### PR TITLE
Issue 6915: Move AWS and DC/OS deployment guides

### LINKED ISSUE #6915: Remove AWS and DC/OS deployment guides
Mesosphere and Terraform deployment strategies are unmaintained and unsupported, and should be removed.

https://pravega.io/docs/latest/deployment/dcos-install/
https://pravega.io/docs/latest/deployment/aws-install/

### PR DESCRIPTION
**Pull request by derekm**
**Change log description**  
Remove AWS Terraform and DC/OS Mesosphere deployment guides

**Purpose of the change**  
Fixes #6915

### CODE REVIEW

File: documentation/src/docs/deployment/deployment.md [hunk 1]
```diff
@@ -35,9 +35,9 @@ The following prerequisites are required for running Pravega in all modes.
 
 The following prerequisites are required for running in production. These are only required for running in distributed mode.
 
-- External HDFS 2.7
-- Zookeeper 3.6.1
-- Bookkeeper 4.11.1
+- AWS S3, Azure Blob Storage, Google Cloud Storage or a mounted distributed filesystem
```
Comment (RaulGracia) [inline documentation/src/docs/deployment/deployment.md:38]: Maybe we can leave HDFS as the last option here, right? We still support that.

**Review by RaulGracia — CHANGES_REQUESTED** [review]
Minor comments. But there is a more relevant topic: if we are deleting this old documentation, what should we do with https://github.com/pravega/pravega/tree/master/deployment/aws package? Should we delete that as well?

**Comment by derekm** [comment]
> if we are deleting this old documentation, what should we do with https://github.com/pravega/pravega/tree/master/deployment/aws package? Should we delete that as well?

I know people do still use Terraform. Even if this is unmaintained, it probably is a good start for someone who wants to pick it back up again and bring it up-to-date, supposing a Terraform user comes to Pravega.

How should we best handle unmaintained but resurrect-able code? Delete and then remember/recover when someone asks? Or let it sit stale until someone notices?

If we choose to let it sit stale, maybe I should move the docs out of the way to deployment/aws and deployment/dcos instead of delete them.

**Review by RaulGracia — APPROVED** [review]

**Review by anishakj — APPROVED** [review]
LGTM

**Pull request merged**